src/device/usbd.c

```diff
@@ -727,41 +703,59 @@ static bool process_set_config(uint8_t rhport, uint8_t cfg_num)
 
   while( p_desc < desc_end )
   {
-    // Each interface always starts with Interface or Association descriptor
+    tusb_desc_interface_assoc_t const * desc_itf_assoc = NULL;
+
+    // Class will always starts with Interface Association (if any) and then Interface descriptor
     if ( TUSB_DESC_INTERFACE_ASSOCIATION == tu_desc_type(p_desc) )
     {
-      p_desc = tu_desc_next(p_desc); // ignore Interface Association
-    }else
-    {
-      TU_ASSERT( TUSB_DESC_INTERFACE == tu_desc_type(p_desc) );
+      desc_itf_assoc = (tusb_desc_interface_assoc_t const *) p_desc;
+      p_desc = tu_desc_next(p_desc); // next to Interface
+    }
 
-      tusb_desc_interface_t const * desc_itf = (tusb_desc_interface_t const*) p_desc;
+    TU_ASSERT( TUSB_DESC_INTERFACE == tu_desc_type(p_desc) );
 
-      uint8_t drv_id;
-      uint16_t drv_len;
+    tusb_desc_interface_t const * desc_itf = (tusb_desc_interface_t const*) p_desc;
+    uint8_t drv_id;
+    uint16_t drv_len;
 
-      for (drv_id = 0; drv_id < USBD_CLASS_DRIVER_COUNT; drv_id++)
+    for (drv_id = 0; drv_id < USBD_CLASS_DRIVER_COUNT; drv_id++)
+    {
+      usbd_class_driver_t const *driver = &_usbd_driver[drv_id];
+
+      drv_len = 0;
+      if ( driver->open(rhport, desc_itf, &drv_len) )
```

Comment:
Should the IAD be passed to ```open```, or the first interface?

I don't think it matters so much since USB classes are not defined with a variable number of interfaces?

Reply:
It doesn't make much difference for class driver since they already know what to do. It is mostly for host for managing and dynamic driver binding.  not all class support IAD, since we use open() as driver check now, best to keep it simple for all. We can always change later on if needed.